aiida/orm/entities.py

```diff
@@ -144,25 +160,32 @@ def find(self, filters=None, order_by=None, limit=None):
         :return: a list of resulting matches
         :rtype: list
         """
-        query = self.query()
-        filters = filters or {}
-        query.add_filter(self.entity_type, filters)
-        if order_by:
-            query.order_by({self.entity_type: order_by})
-        if limit:
-            query.limit(limit)
-
+        query = self.query(filters=filters, order_by=order_by, limit=limit)
         return [_[0] for _ in query.all()]
 
     def all(self):
         """
         Get all entities in this collection
 
-        :return: A collection of users matching the criteria
+        :return: A list of all entities
         :rtype: list
         """
         return [_[0] for _ in self.query().all()]
 
+    def count(self, filters=None, limit=None, offset=None):
```

Comment:
do `limit` and `offset` make sense in the context of `count`? Couldn't you just do `max(Entity.objects.count() - offset, limit)` if you really wanted to bound the answer? Or is this for efficiency reasons?

Reply:
I guess one could, but at the same time, since `limit` and `offset` are already implemented in the `QueryBuilder` and they are not mandatory for `.objects.count()`, it doesn't seem to be a big issue.
The only problem I can see is a definition question of why the `QueryBuilder` `count()` method does not have similar parameter definition as `.objects.count()` - however, I see this as being fine, since I view the two methods to be philosophically different (although both are `QueryBuilder`s).
In the end, I prefer the elegance of this implementation (`Entity.objects.count(offset=offset, limit=limit)`), instead of `max(Entity.objects.count() - offset, limit)` :sweat_smile:

Reply:
> The only problem I can see is a definition question of why the `QueryBuilder` `count()` method does not have a similar parameter definition as `.objects.count()`

`QueryBuilder.count` does not have those arguments because they don't really make sense for that function.

> In the end, I prefer the elegance of this implementation (`Entity.objects.count(offset=offset, limit=limit)`), instead of `max(Entity.objects.count() - offset, limit)`

What is the point of `Entity.objects.count(offset=offset, limit=limit)` though. Why would one ever want to do that?

Reply:
> What is the point of `Entity.objects.count(offset=offset, limit=limit)` though. Why would one ever want to do that?

Well --- now that you put it that way (and I actually am forced to think it through) you do make a lot of sense. Hmm. Limit definitely doesn't make sense. Offset, sort of, although it is as simple to take the difference with the total count, for sure. Hmm. Yes, I see. Silly of me.
There shouldn't be any large optimization benefits of doing it like this either, I think? ... I will remove them as parameters.